x/txfees/keeper/feedecorator_test.go

```diff
@@ -19,137 +21,98 @@ func (suite *KeeperTestSuite) TestFeeDecorator() {
+	for isCheckTx := 0; isCheckTx <= 1; isCheckTx++ {
+		tests = append(tests, []testcase{
+			{
+				name:       fmt.Sprintf("no min gas price - %s. Fails w/ consensus minimum", txType[isCheckTx]),
+				txFee:      sdk.NewCoins(),
+				isCheckTx:  isCheckTx == 1,
+				expectPass: false,
+			},
+			{
+				name:       fmt.Sprintf("LT Consensus min gas price - %s", txType[isCheckTx]),
+				txFee:      sdk.NewCoins(sdk.NewInt64Coin(baseDenom, consensusMinFeeAmt-1)),
+				isCheckTx:  isCheckTx == 1,
+				expectPass: false,
+			},
+			{
+				name:       fmt.Sprintf("Consensus min gas price - %s", txType[isCheckTx]),
+				txFee:      sdk.NewCoins(sdk.NewInt64Coin(baseDenom, consensusMinFeeAmt)),
+				isCheckTx:  isCheckTx == 1,
+				expectPass: true,
+			},
+			{
+				name:       fmt.Sprintf("multiple fee coins - %s", txType[isCheckTx]),
+				txFee:      sdk.NewCoins(sdk.NewInt64Coin(baseDenom, 1), sdk.NewInt64Coin(uion, 1)),
+				isCheckTx:  isCheckTx == 1,
+				expectPass: false,
+			},
+			{
+				name:  fmt.Sprintf("%s work with insufficient mempool fee in %s", succesType[isCheckTx], txType[isCheckTx]),
+				txFee: sdk.NewCoins(sdk.NewInt64Coin(baseDenom, consensusMinFeeAmt)), // consensus minimum
+				minGasPrices: sdk.NewDecCoins(sdk.NewDecCoinFromDec(baseDenom,
+					sdk.MustNewDecFromStr("0.1"))),
+				isCheckTx:  isCheckTx == 1,
+				expectPass: isCheckTx != 1,
+			},
+			{
+				name:  fmt.Sprintf("%s work with insufficient converted mempool fee in %s", succesType[isCheckTx], txType[isCheckTx]),
+				txFee: sdk.NewCoins(sdk.NewInt64Coin(uion, 25)), // consensus minimum
+				minGasPrices: sdk.NewDecCoins(sdk.NewDecCoinFromDec(baseDenom,
+					sdk.MustNewDecFromStr("0.1"))),
+				isCheckTx:  isCheckTx == 1,
+				expectPass: isCheckTx != 1,
+			},
+			{
+				name:       "invalid fee denom",
+				txFee:      sdk.NewCoins(sdk.NewInt64Coin("moooooo", 1000)),
+				isCheckTx:  isCheckTx == 1,
+				expectPass: false,
+			},
+		}...)
+	}
+
+	custTests := []testcase{
 		{
 			name:  "works with valid basedenom fee",
 			txFee: sdk.NewCoins(sdk.NewInt64Coin(baseDenom, 1000)),
 			minGasPrices: sdk.NewDecCoins(sdk.NewDecCoinFromDec(baseDenom,
 				sdk.MustNewDecFromStr("0.1"))),
-			gasRequested: 10000,
-			isCheckTx:    true,
-			expectPass:   true,
-			baseDenomGas: true,
-		},
-		{
-			name:  "doesn't work with not enough fee in checktx",
-			txFee: sdk.NewCoins(sdk.NewInt64Coin(baseDenom, 1)),
-			minGasPrices: sdk.NewDecCoins(sdk.NewDecCoinFromDec(baseDenom,
-				sdk.MustNewDecFromStr("0.1"))),
-			gasRequested: 10000,
-			isCheckTx:    true,
-			expectPass:   false,
-			baseDenomGas: true,
-		},
-		{
-			name:  "works with not enough fee in delivertx",
-			txFee: sdk.NewCoins(sdk.NewInt64Coin(baseDenom, 1)),
-			minGasPrices: sdk.NewDecCoins(sdk.NewDecCoinFromDec(baseDenom,
-				sdk.MustNewDecFromStr("0.1"))),
-			gasRequested: 10000,
-			isCheckTx:    false,
-			expectPass:   true,
-			baseDenomGas: true,
+			isCheckTx:  true,
+			expectPass: true,
```

Comment:
nit: I feel like the top two tests should be a part of array above, not cusTests